packages/build-scripts/getBaseConfig.ts

```diff
@@ -32,11 +31,25 @@ export function getBaseConfig(platform: Platform, formats: Format[], _options: O
                               if (format === 'iife') {
                                   options.define = {
                                       ...options.define,
-                                      __DEV__: `${isDebugBuild}`,
+                                      'import.meta.env.MODE': `"${isDebugBuild ? 'development' : 'production'}"`,
+                                      'process.env.NODE_ENV': `"${isDebugBuild ? 'development' : 'production'}"`,
                                   };
                                   options.target = BROWSERSLIST_TARGETS;
+                              } else {
+                                  options.banner = {
+                                      ...options.banner,
+                                      js: [
+                                          options.banner?.js,
+                                          ';',
+                                          'var __DEV__ = ',
+                                          "process.env.NODE_ENV === 'development'",
```

Comment:
I think this will still throw when process is undefined. Would process?.env work or can we not use that sort of syntax here?

Reply:
It also throws in plain Node, though I'm not sure if that's representative of a real issue:

```
var __DEV__ = process.env.NODE_ENV === 'development' || import.meta.env.MODE === 'development';
                                                               ^^^^
Uncaught SyntaxError: Cannot use 'import.meta' outside a module
```

Reply:
Yeah, this needs a whole lotta guarding, I imagine, but it's harder than I'd like, because as soon as I do this:

```ts
typeof process !== 'undefined' && process.env.NODE_ENV === 'development'
```

The `define` rule stops working and you get this:

```ts
undefined === 'development'
```

…or some garbage and no DCE takes place.

Reply:
Related to our interests: https://github.com/terser/terser/issues/1536